# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/isdn4k-utils/isdn.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./iprofd/iprofd.c.orig	Fri Jun 29 11:08:29 2001
+++ ./iprofd/iprofd.c	Fri Jun 29 11:08:29 2001
@@ -62,6 +62,11 @@
 #include <signal.h>
 #include <linux/isdn.h>
 
+/* Ugly hack, but the isdn header now defines it in kernel space. */
+#ifndef ISDN_MAX_CHANNELS
+#  define ISDN_MAX_CHANNELS 64
+#endif
+
 typedef unsigned char uchar;
 
 int isdnctrl_fd;
--- ./isdnlog/isdnlog/isdnlog.h.orig    Fri Dec 31 05:30:02 1999
+++ ./isdnlog/isdnlog/isdnlog.h Fri Jun 29 11:38:15 2001
@@ -284,6 +284,14 @@
 #include <rate.h>
 #include "socket.h"
 
+/* Ugly hack, but the isdn header now defines it in kernel space. */
+#ifndef ISDN_MAX_CHANNELS
+#  define ISDN_MAX_CHANNELS 64
+#endif
+#ifndef ISDN_MAX_DRIVERS
+#  define ISDN_MAX_DRIVERS 32
+#endif
+
 /****************************************************************************/
 
 #define MAX_CALLS_IN_QUEUE    100
--- ./xisdnload/xisdnload.c.orig	Wed Jun 23 06:01:36 1999
+++ ./xisdnload/xisdnload.c	Fri Jun 29 11:13:42 2001
@@ -60,6 +60,14 @@
 
 #include "xisdnload.bit"
 
+/* Ugly hack, but the isdn header now defines it in kernel space. */
+#ifndef ISDN_MAX_CHANNELS
+#  define ISDN_MAX_CHANNELS 64
+#endif
+#ifndef ISDN_MAX_DRIVERS
+#  define ISDN_MAX_DRIVERS 32
+#endif
+
 char *ProgramName;
 
 static void quit();
--- ./imontty/imontty.c.orig	Thu May 11 03:55:14 2000
+++ ./imontty/imontty.c	Fri Jun 29 11:14:07 2001
@@ -12,6 +12,14 @@
 #include <linux/isdn.h>
 #include "imontty.h"
 
+/* Ugly hack, but the isdn header now defines it in kernel space. */
+#ifndef ISDN_MAX_CHANNELS
+#  define ISDN_MAX_CHANNELS 64
+#endif
+#ifndef ISDN_MAX_DRIVERS
+#  define ISDN_MAX_DRIVERS 32
+#endif
+
 struct phone_entry {
    struct phone_entry *next;
    char phone[30];
--- ./imon/imon.c.orig	Sat May 17 05:23:35 1997
+++ ./imon/imon.c	Fri Jun 29 11:14:21 2001
@@ -56,6 +56,11 @@
 
 #include <linux/isdn.h>
 
+/* Ugly hack, but the isdn header now defines it in kernel space. */
+#ifndef ISDN_MAX_CHANNELS
+#  define ISDN_MAX_CHANNELS 64
+#endif
+
 #define KEY_Q                81
 #define KEY_q               113
 
--- ./xmonisdn/Net.c.orig       Tue Aug  3 07:00:20 1999
+++ ./xmonisdn/Net.c    Fri Jun 29 12:03:25 2001
@@ -49,6 +49,11 @@
 #include <X11/Xmu/Drawing.h>
 #include <X11/extensions/shape.h>
 
+/* Ugly hack, but the isdn header now defines it in kernel space. */
+#ifndef ISDN_MAX_CHANNELS
+#  define ISDN_MAX_CHANNELS 64
+#endif
+
 /* Only allow calling scripts when setuid root if the -r option is given */
 extern int allow_setuid;
 /*
